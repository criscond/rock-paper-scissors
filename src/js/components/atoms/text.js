import { createElement } from '../services';

const text = ({ htmlTag, value, className = '' }) => {

  const element = createElement({ htmlTag });
  element.className = className;
  element.innerHTML = value;

  return element;
}

export default text;
