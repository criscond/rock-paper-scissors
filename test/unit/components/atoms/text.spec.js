import text from '../../../../src/js/components/atoms/text';
describe('Text', () => {

  it('should create a text element', () => {
    const data = {
      htmlTag: 'h1',
      value: 'Hello'
    }
    const element = text(data);

    expect(element).toMatchSnapshot();
  });

  it('should create a text element with class', () => {
    const data = {
      ...data,
      className: 'h1-class'
    }
    const element = text(data);

    expect(element.className).toEqual(data.className);
  });
});
