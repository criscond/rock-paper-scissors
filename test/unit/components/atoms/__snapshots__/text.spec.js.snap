// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should create a text element 1`] = `
<h1
  class=""
>
  Hello
</h1>
`;
